### PR TITLE
DSD-698: Removing BEM references in the repo

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-698](https://jira.nypl.org/browse/DSD-698)

## This PR does the following:

- Removes the BEM function and its use in the repo.
- Updates the `Placeholder` (only used in the `Template` story) and `Modal` components by removing the BEM use in them. Updates the `modifiers` prop in `Placeholder` to `modifier` since now it's just a simple string for the class name.

## How has this been tested?

By running tests, building the repo, and running storybook.

## Accessibility concerns or updates

-N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr838-aaboiougiktlm1uncr8joikd7eemiqrd.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61d8af97dc579fc65d9eee28

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr838-aaboiougiktlm1uncr8joikd7eemiqrd.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61d8af97dc579fc65d9eee28

File: src/components/Placeholder/Placeholder.tsx [hunk 1]
```diff
@@ -1,24 +1,19 @@
 import * as React from "react";
-import bem from "../../utils/bem";
 
 export interface PlaceholderProps {
-  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
-  blockName?: string;
-  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
-  modifiers?: string[];
+  /** Modifier string for the classname. */
+  modifier?: string;
```
Comment (bigfishdesign13) [inline src/components/Placeholder/Placeholder.tsx]: Should we just call this `variant`?

**Review by bigfishdesign13 — COMMENTED** [review]
I have just one question regarding the props for `Placeholder`.

File: src/components/Placeholder/Placeholder.tsx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/Placeholder/Placeholder.tsx]: Yeah, that works. Do we need this component long-term?

Comment (bigfishdesign13) [reply src/components/Placeholder/Placeholder.tsx]: `Placeholder` is an internal component, so it'll be needed as long as we want to use it.  Right now, it is only used in the stories and tests for `Template`, where I think it is useful.  Now that BEM has been removed, I don't think it needs any further development.

**Review by bigfishdesign13 — APPROVED** [review]
All good.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr838-aaboiougiktlm1uncr8joikd7eemiqrd.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61d8af97dc579fc65d9eee28

**Pull request merged**